apollo-ios-pagination/Sources/ApolloPagination/GraphQLQueryPager.swift

```diff
@@ -37,22 +60,27 @@ public class GraphQLQueryPager<InitialQuery: GraphQLQuery, PaginatedQuery: Graph
     client: ApolloClientProtocol,
     initialQuery: InitialQuery,
     extractPageInfo: @escaping (PageExtractionData) -> P,
-    nextPageResolver: @escaping (P) -> PaginatedQuery
+    nextPageResolver: ((P) -> PaginatedQuery)?,
+    previousPageResolver: ((P) -> PaginatedQuery)?
   ) {
     pager = .init(
       client: client,
       initialQuery: initialQuery,
       extractPageInfo: extractPageInfo,
-      nextPageResolver: nextPageResolver
+      nextPageResolver: nextPageResolver,
+      previousPageResolver: previousPageResolver
     )
     Task {
-      let varMapPublisher = await pager.$varMap
+      let previousPageVarMapPublisher = await pager.$previousPageVarMap
+      let nextPageVarMapPublisher = await pager.$nextPageVarMap
       let initialPublisher = await pager.$initialPageResult
-      varMapPublisher.combineLatest(initialPublisher).sink { [weak self] _ in
+      previousPageVarMapPublisher.combineLatest(initialPublisher, nextPageVarMapPublisher).sink { [weak self] _ in
         guard let self else { return }
         Task {
-          let value = await self.pager.pageTransformation()
-          self.canLoadNextSubject.send(value?.canLoadMore ?? false)
+          let canLoadNext = await self.pager.canLoadNext
+          let canLoadPrevious = await self.pager.canLoadPrevious
```

Comment:
Same as above, [actor isolation](https://github.com/apple/swift-evolution/blob/main/proposals/0313-actor-isolation-control.md) could allow us to only have a single await here if the `Actor` has a property that returned both of these.

```swift
/// on Actor
var canLoadPages: (next: Bool, previous: Bool) {
    (canLoadNext, canLoadPrevious)
}
```